Mathlib/RingTheory/Polynomial/Resultant/Basic.lean

```diff
@@ -58,12 +59,50 @@ variable (f g : R[X]) (m n : ℕ)
     rw [sylvester, Matrix.of_apply, Fin.addCases_right, Matrix.diagonal_apply]
     split_ifs <;> simp_all [Fin.ext_iff]
 
-end sylvester
+/--
+The Sylvester matrix for `f` and `f.derivative`, modified by dividing the bottom row by
+the leading coefficient of `f`. Important because its determinant is (up to a sign) the
+discriminant of `f`.
+-/
+noncomputable def
+sylvester_deriv (f : R[X]) : Matrix (Fin (2 * f.natDegree - 1)) (Fin (2 * f.natDegree - 1)) R :=
```

Comment:
Do you think this would be easier if `2*f.natDegree-1` is replaced with `f.natDegree+(f.natDegree-1)`?

Reply:
Maybe it would. Somehow I had a notion that `2 * f.natDegree - 1` was more "canonical" in some sense, but in retrospect I'm not sure that I can justify that, and it would be nice to avoid some of the `finCongr`s.